conans/client/build/cmake_flags.py

```diff
@@ -279,68 +279,63 @@ def get_definitions(self):
         runtime = self._ss("compiler.runtime")
         build_type = self._ss("build_type")
 
-        ret = OrderedDict()
-        ret.update(runtime_definition(runtime))
+        defines = OrderedDict([
+            ("CONAN_EXPORTED", "1"),
+            # Disable CMake export registry #3070 (CMake installing modules in user home's)
+            ("CMAKE_EXPORT_NO_PACKAGE_REGISTRY", "ON")
+        ])
 
         if self._forced_build_type and self._forced_build_type != build_type:
             self._output.warn("Forced CMake build type ('%s') different from the settings build "
                               "type ('%s')" % (self._forced_build_type, build_type))
             build_type = self._forced_build_type
 
-        ret.update(build_type_definition(build_type, self._generator))
-
         if str(os_) == "Macos":
             if arch == "x86":
-                ret["CMAKE_OSX_ARCHITECTURES"] = "i386"
-
-        ret.update(self._cmake_cross_build_defines())
-        ret.update(self._get_cpp_standard_vars())
-
-        ret["CONAN_EXPORTED"] = "1"
-        ret.update(in_local_cache_definition(self._conanfile.in_local_cache))
+                defines["CMAKE_OSX_ARCHITECTURES"] = "i386"
 
         if compiler:
-            ret["CONAN_COMPILER"] = compiler
+            defines["CONAN_COMPILER"] = compiler
         if compiler_version:
-            ret["CONAN_COMPILER_VERSION"] = str(compiler_version)
+            defines["CONAN_COMPILER_VERSION"] = str(compiler_version)
 
         # C, CXX, LINK FLAGS
         if compiler == "Visual Studio":
             if self._parallel:
                 flag = parallel_compiler_cl_flag(output=self._output)
-                ret['CONAN_CXX_FLAGS'] = flag
-                ret['CONAN_C_FLAGS'] = flag
+                defines["CONAN_CXX_FLAGS"] = flag
+                defines["CONAN_C_FLAGS"] = flag
         else:  # arch_flag is only set for non Visual Studio
             arch_flag = architecture_flag(compiler=compiler, os=os_, arch=arch)
             if arch_flag:
-                ret['CONAN_CXX_FLAGS'] = arch_flag
-                ret['CONAN_SHARED_LINKER_FLAGS'] = arch_flag
-                ret['CONAN_C_FLAGS'] = arch_flag
+                defines["CONAN_CXX_FLAGS"] = arch_flag
```

Comment:
Those are not really defines, nor preprocessor definitions. They are pure cmake variables, so if you would like to rename ``ret`` to something more meaningful (I agree it would be a good thing), I would consider something like ``cmake_variables``.